packages/core/src/UserState.ts

```diff
@@ -360,545 +289,192 @@ export default class UserState extends Synchronizer {
+                pos_rep: posRep,
+                neg_rep: negRep,
+                graffiti,
+                timestamp,
+                new_pos_rep: epochKeyData.map(({ posRep }) => posRep),
+                new_neg_rep: epochKeyData.map(({ negRep }) => negRep),
+                new_graffiti: epochKeyData.map(({ graffiti }) => graffiti),
+                new_timestamp: epochKeyData.map(({ timestamp }) => timestamp),
+                epoch_tree_elements: epochKeyData.map(({ proof }) => proof),
+                epoch_tree_root: epochTree.root,
+            }
             const results = await this.prover.genProofAndPublicSignals(
-                Circuit.processAttestations,
-                processAttestationCircuitInputs[i]
+                Circuit.userStateTransition,
+                stringifyBigInts(circuitInputs)
             )
-            processAttestationProofs.push(
-                new ProcessAttestationsProof(
-                    results.publicSignals,
-                    results.proof,
-                    this.prover
-                )
-            )
-        }
 
-        const finalProofResults = await this.prover.genProofAndPublicSignals(
-            Circuit.userStateTransition,
-            finalTransitionCircuitInputs
-        )
-
-        return {
-            startTransitionProof: new StartTransitionProof(
-                startTransitionresults.publicSignals,
-                startTransitionresults.proof,
-                this.prover
-            ),
-            processAttestationProofs: processAttestationProofs,
-            finalTransitionProof: new UserTransitionProof(
-                finalProofResults.publicSignals,
-                finalProofResults.proof,
+            return new UserStateTransitionProof(
+                results.publicSignals,
+                results.proof,
                 this.prover
-            ),
+            )
         }
-    }
 
     /**
      * Generate a reputation proof of current user state and given conditions
-     * @param attesterId The attester ID that the user wants to proof the reputation
      * @param epkNonce The nonce determines the output of the epoch key
      * @param minRep The amount of reputation that user wants to prove. It should satisfy: `posRep - negRep >= minRep`
-     * @param proveGraffiti The boolean flag that indicates if user wants to prove graffiti pre-image
-     * @param graffitiPreImage The pre-image of the graffiti
-     * @param spendAmount The amount of reputation to spend.
-     * In the circuit, it will compute the reputation nullifiers of the given nonce. If the reputation nullifier is used
-     * to spend reputation, it cannot be spent again.
      * @returns The reputation proof of type `ReputationProof`.
      */
     public genProveReputationProof = async (
-        attesterId: BigInt,
         epkNonce: number,
-        minRep?: number,
-        proveGraffiti?: BigInt,
-        graffitiPreImage?: BigInt,
```

Comment:
should prove the graffiti preimage

Reply:
[27a46fc](https://github.com/Unirep/Unirep/pull/161/commits/27a46fcad3bba01436b2647d28825e0ee1f57143) fixed